lib/services/elasticsearch.js

```diff
@@ -1231,13 +1247,33 @@ class ElasticSearch extends Service {
       });
   }
 
+  _mergeDefaultMapping (index, oldMapping, newMapping) {
+    const collectionMapping = {};
+    const commonMapping = this.kuzzle.indexCache.defaultMappings[index] || this.config.commonMapping;
+
+    collectionMapping.dynamic = newMapping.dynamic || oldMapping.dynamic || this.config.dynamic;
+    collectionMapping._meta = newMapping._meta || oldMapping._meta || {};
+
+    // Inject kuzzle meta but preserve existing ones
+    if (oldMapping.properties && oldMapping.properties._kuzzle_info) {
+      Object.assign(
+        commonMapping._kuzzle_info.properties,
+        oldMapping.properties._kuzzle_info.properties
+      );
+    }
+
+    collectionMapping.properties = Object.assign({}, newMapping.properties, commonMapping);
```

Comment:
I'm sorry, I'm still having a hard time figuring out how this could work (I suggest you write a jsDoc for this function in order to properly explain its purpose)...

This like makes me tick because here, the properties of newMapping will be overwritten by the ones that have the same key in commonMapping.
This makes that the action performed by this function is not a "merge" but rather "add some new properties that are not specified in the default mapping nor the already existing one".

So, is this correct? Am I missing something?

Reply:
> This like makes me tick because here, the properties of newMapping will be overwritten by the ones that have the same key in commonMapping.

It's only for the `_kuzzle_info` field, not for every mapping property.  
This method can be called either: 
 * at the first creation of a collection : we need to apply the index default mapping
 * when a collection mapping is updated : we need to protect the index common mapping